discovery-provider/src/tasks/playlists.py

```diff
@@ -40,23 +42,28 @@ def playlist_state_update(
                         "events": [],
                     }
 
-                playlist_events_lookup[playlist_id]["events"].append(event_type)
-
-                playlist_events_lookup[playlist_id]["playlist"] = parse_playlist_event(
+                playlist_record = parse_playlist_event(
                     self,
                     update_task,
                     entry,
                     event_type,
                     playlist_events_lookup[playlist_id]["playlist"],
                     block_timestamp,
+                    session
                 )
 
-            num_total_changes += len(playlist_events_tx)
+                if playlist_record is not None:
+                    playlist_events_lookup[playlist_id]["events"].append(event_type)
+                    playlist_events_lookup[playlist_id]["playlist"] = playlist_record
+                    processedEntries += 1
+
+            num_total_changes += processedEntries
 
     for playlist_id, value_obj in playlist_events_lookup.items():
         logger.info(f"playlists.py | Adding {value_obj['playlist']})")
-        invalidate_old_playlist(session, playlist_id)
-        session.add(value_obj["playlist"])
+        if value_obj["events"]:
```

Comment:
@vicky-g do we still need this `if` check after the `is not None` check above?

Reply:
yeah cus in line 40-42, for that `playlist_id` key, the `events` is it init'd to `[]`. then, if there's only one event and it's a blacklisted event, then the line 64 makes sure we don't actually try to add an empty event to the session